core/domain/exp_jobs_one_off.py

```diff
@@ -470,3 +473,61 @@ def map(item):
     @staticmethod
     def reduce(key, values):
         yield (key, values)
+
+
+class ExplorationContentValidationJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that checks the html content of an exploration and validates it.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        exploration = exp_services.get_exploration_from_model(item)
+        err_dict = {}
+
+        with open('./core/domain/htmlValidation.yaml', 'r') as fp:
+            htmlValidation = yaml.load(fp)
+            allowed_parent_list = htmlValidation['allowed_parent_list']
+            allowed_tag_list = htmlValidation['allowed_tag_list']
+
+        for state in exploration.states.itervalues():
+            result = exp_services.find_all_values_for_key(
+                'html', state.to_dict())
+
+            for html_data in result:
+                html_data = html_data.encode('utf-8')
+                soup = BeautifulSoup(html_data, 'html.parser')
+
+                # Checking for tags not allowed in RTE.
+                for tag in soup.findAll():
+                    if tag.name not in allowed_tag_list:
+                        if 'invalidTags' in err_dict:
+                            err_dict['invalidTags'] += [tag.name]
+                        else:
+                            err_dict['invalidTags'] = [tag.name]
+
+                # Checking for parent-child relation that are not
+                # allowed in RTE.
+                for key, value in allowed_parent_list.iteritems():
```

Comment:
I suggest you either comment describing what `key` and `value` are here, or rename those vars to something more descriptive.

Edit: is value here a list or dictionary? I can't tell.

Reply:
Updated!